pydoctor/test/test_astbuilder.py

```diff
@@ -1188,21 +1300,58 @@ def __init__(self):
     assert ann_str_and_line(mod.contents['m']) == ('bytes', 19)
 
 @systemcls_param
-def test_type_from_attrib(systemcls: Type[model.System]) -> None:
+def test_attrs_attrib_type(systemcls: Type[model.System]) -> None:
+    """An attr.ib's "type" or "default" argument is used as an alternative
+    type annotation.
+    """
     mod = fromText('''
     import attr
     from attr import attrib
+    @attr.s
     class C:
         a = attr.ib(type=int)
         b = attrib(type=int)
         c = attr.ib(type='C')
         d = attr.ib(default=True)
+        e = attr.ib(123)
     ''', modname='test', systemcls=systemcls)
     C = mod.contents['C']
     assert type2str(C.contents['a'].annotation) == 'int'
     assert type2str(C.contents['b'].annotation) == 'int'
     assert type2str(C.contents['c'].annotation) == 'C'
     assert type2str(C.contents['d'].annotation) == 'bool'
+    assert type2str(C.contents['e'].annotation) == 'int'
+
+@systemcls_param
+def test_attrs_attrib_instance(systemcls: Type[model.System]) -> None:
+    """An attr.ib attribute is classified as an instance variable."""
+    mod = fromText('''
+    import attr
+    @attr.s
+    class C:
+        a = attr.ib(type=int)
+    ''', modname='test', systemcls=systemcls)
+    C = mod.contents['C']
+    assert C.contents['a'].kind == 'Instance Variable'
+
+@systemcls_param
+def test_attrs_auto_instance(systemcls: Type[model.System]) -> None:
+    """Attrs auto-attributes are classified as instance variables."""
+    mod = fromText('''
+    from typing import ClassVar
+    import attr
+    @attr.s(auto_attribs=True)
+    class C:
+        a: int
+        b: bool = False
+        c: ClassVar[str]  # explicit class variable
+        d = 123  # ignored by auto_attribs because no annotation
+    ''', modname='test', systemcls=systemcls)
+    C = mod.contents['C']
+    assert C.contents['a'].kind == 'Instance Variable'
+    assert C.contents['b'].kind == 'Instance Variable'
+    assert C.contents['c'].kind == 'Class Variable'
+    assert C.contents['d'].kind == 'Class Variable'
```

Comment:
[`attr.s`](https://www.attrs.org/en/stable/api.html#attr.s) itself will ignore unannotated fields:

> Attributes annotated as `typing.ClassVar`, and attributes that are neither annotated nor set to an `attr.ib` are **ignored**.

So I think we should treat `d` the same as when `auto_attribs` is not present, which is defaulting to class variable. Note that if there is any assignment to `self.d`, we automatically convert to instance variable at that point.

Reply:
>  Note that if there is any assignment to `self.d`, we automatically convert to instance variable at that point.

I think that it is worth adding a new test to document this behaviour.

Reply:
This is tested as part of `test_inline_docstring_instancevar`.

I don't think testing the combination of `auto_attribs=True` and assignment to `self.d` is useful, since whether the `attrs` support handles it correctly or not, the result is always "Instance Variable". In other words, I can't think of a bug that would fail the combination test but pass the individual tests.